Fails to build without verbs

I am building a copy in a virtual machine with Ubuntu 17.04 and it is failing to build for me. It should build fine without verbs but apparently not:

```
./configure --prefix=/opt/ucx --with-rdmacm=no --with-verbs=no --with-xpmem=/opt/xpmem && make
...
make[2]: Entering directory '/home/hjelmn/src/ucx/src/uct'
  CC       base/libuct_la-uct_md.lo
  CC       base/libuct_la-uct_mem.lo
  CC       base/libuct_la-uct_iface.lo
  CC       base/libuct_la-uct_worker.lo
  CC       tcp/libuct_la-tcp_ep.lo
  CC       tcp/libuct_la-tcp_iface.lo
  CC       tcp/libuct_la-tcp_md.lo
  CC       tcp/libuct_la-tcp_net.lo
In file included from /home/hjelmn/src/ucx/src/uct/ib/base/ib_device.h:10:0,
                 from tcp/tcp_net.c:13:
/home/hjelmn/src/ucx/src/uct/ib/base/ib_verbs.h:14:30: fatal error: infiniband/verbs.h: No such file or directory
 #include <infiniband/verbs.h>
                              ^
compilation terminated.

```

--- a/src/uct/tcp/tcp_net.c
+++ b/src/uct/tcp/tcp_net.c
@@ -10,7 +10,9 @@
 #  include "config.h"
 #endif
 
+#if HAVE_IB
 #include <uct/ib/base/ib_device.h> /* for ipoib header size */
+#endif
 #include <ucs/sys/string.h>
 #include <linux/sockios.h>
 #include <linux/types.h>
@@ -91,6 +93,7 @@ ucs_status_t uct_tcp_netif_caps(const char *if_name, double *latency_p,
                      ETH_FCS_LEN + /* CRC */
                      12; /* inter-packet gap */
         break;
+#if HAVE_IB
     case ARPHRD_INFINIBAND:
         ll_headers = UCT_IB_LRH_LEN +
                      UCT_IB_GRH_LEN +
@@ -101,6 +104,7 @@ ucs_status_t uct_tcp_netif_caps(const char *if_name, double *latency_p,
                      UCT_IB_VCRC_LEN +
                      UCT_IB_DELIM_LEN;
         break;
+#endif
     default:
         ll_headers = 0;
         break;

